features/measurements/impl/src/main/java/org/opennms/netmgt/measurements/impl/JEXLExpressionEngine.java

```diff
@@ -122,6 +126,12 @@ public void applyExpressions(final QueryRequest request, final FetchResults resu
         jexlValues.put("__inf", Double.POSITIVE_INFINITY);
         jexlValues.put("__neg_inf", Double.NEGATIVE_INFINITY);
         jexlValues.put("NaN", Double.NaN);
+        
+        // add context as a reference to be picked up by complex formulae
```

Comment:
Can we introduce a new `eval` function (or similar) instead of exposing the engine and context as variables?

We shouldn't allow scripts to interface with the engine directly.